prelude/toolchains/cxx.bzl

```diff
@@ -170,49 +203,54 @@ def _system_cxx_toolchain_impl(ctx: AnalysisContext):
         CxxPlatformInfo(name = "x86_64"),
     ]
 
-def _windows_linker_wrapper(ctx: AnalysisContext, linker: cmd_args) -> cmd_args:
-    # Linkers pretty much all support @file.txt argument syntax to insert
-    # arguments from the given text file, usually formatted one argument per
-    # line.
-    #
-    # - GNU ld: https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html
-    # - lld is command line compatible with GNU ld
-    # - MSVC link.exe: https://learn.microsoft.com/en-us/cpp/build/reference/linking?view=msvc-170#link-command-files
-    #
-    # However, there is inconsistency in whether they support nesting of @file
-    # arguments inside of another @file.
-    #
-    # We wrap the linker to flatten @file arguments down to 1 level of nesting.
-    return cmd_script(
-        ctx = ctx,
-        name = "windows_linker",
-        cmd = cmd_args(
-            ctx.attrs.linker_wrapper[RunInfo],
-            linker,
-        ),
-        os = ScriptOs("windows"),
-    )
+def _run_info(args):
+    return None if args == None else RunInfo(args = [args])
 
 system_cxx_toolchain = rule(
     impl = _system_cxx_toolchain_impl,
     attrs = {
+        "cxx_tools_info": attrs.default_only(attrs.exec_dep(providers = [CxxToolsInfo], default = select({
+            "DEFAULT": "prelude//toolchains/cxx/clang:path_clang_tools",
+            "config//os:windows": "prelude//toolchains/msvc:msvc_tools",
+        }))),
```

Comment:
So, while using a select to swap between these would normally definitely be the right thing to do, the `system_*_toolchain` rules are a little bit special in that they're supposed to be low-effort defaults that work in somewhat degenerate environments. Relevantly to you, I think for now the expectation probably is that these toolchains should work even while using a platform that does not specify a `config//os:` constraint (as I said, degenerate environments). Now, while I could certainly be convinced that we should change that, I think for the purposes of this PR it's probably better to retain behavior by using a `if host_info` on this rule specifically, instead of a select